### PR TITLE
Fixup goclient upgrade

### PR DESCRIPTION
**Pull request by bethge**
@rebuy-de/prp-kubernetes-deployment - PTAL

Some of the capabilities for decoding manifests have moved to `client-go/kubernetes/schema`.

Separated `app.Render` for easier testability of `scheme.Codecs.UniversalDeserializer().Decode()`.

`pkg/api/render_test.go` is not pretty. On the upside: no added file i/o in the "unit test". Very much open to suggestions.

See:
- https://github.com/kubernetes/client-go/issues/193

### CODE REVIEW

File: pkg/api/render_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import (
+	"github.com/rebuy-de/kubernetes-deployment/pkg/interceptors"
+	"github.com/rebuy-de/kubernetes-deployment/pkg/testutil"
+	"testing"
+)
+
+func TestDecode(t *testing.T) {
+	tcs := map[string]string{
+		"podpreset.yaml": `
```
Comment (svenwltr) [inline pkg/api/render_test.go]: I would put this into an actual file.

File: pkg/api/render.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/rebuy-de/kubernetes-deployment/pkg/templates"
 	log "github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/kubernetes/scheme"
```
Comment (svenwltr) [inline pkg/api/render.go]: Such stable APU. Much wow.

Comment (svenwltr) [reply pkg/api/render.go]: Can you check if this is still required: https://github.com/rebuy-de/kubernetes-deployment/blob/8c23e8d300e9706a2dbe7560411d4a76c9700f04/pkg/api/init.go#L8-L17

Comment (bethge) [reply pkg/api/render.go]: Appears to no longer be needed.

File: pkg/api/render_test.go (hunk 1 above)
Comment (bethge) [reply pkg/api/render_test.go]: Fixed.

File: pkg/api/render_test.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	return string(dat)
+}
+
+func TestDecode(t *testing.T) {
+	dir := "test-fixtures"
+	tcs := map[string]string{
+		"manifest-deployment.yaml": "",
```
Comment (svenwltr) [inline pkg/api/render_test.go]: What do you think about this:

```
"manifest-deployment.yaml": readFile(t, path.Join(dir, fname)),
```

Comment (bethge) [reply pkg/api/render_test.go]: yeah, that works too and is more succinct.

Comment (bethge) [reply pkg/api/render_test.go]: done

**Comment by bethge** [comment]
@rebuy-de/prp-kubernetes-deployment - PTAL

**Pull request merged**